src/selectors/getPrevRank.ts

```diff
@@ -16,8 +16,11 @@ const getPrevRank = (state: State, context: Context, { aboveMeta }: { aboveMeta?
     ? 0
     : children.findIndex(child => !isFunction(child.value))
 
+  // there could be no visible thoughts in the context
+  const noVisibleChildren = i === -1
+
   // between last metaprogramming attribute and first visible child
-  if (i === 0) return children[0].rank - 1
+  if (i === 0 || noVisibleChildren) return children[0].rank - 1
```

Comment:
This is the correct behavior when `aboveMeta` is true. However if `aboveMeta` is false, the new rank should not be above hidden thought. I will add tests to clarify!

Reply:
Got it!